plugin.video.montreal.greek-tv/resources/lib/youtube.py

```diff
@@ -0,0 +1,352 @@
+                try:
+                    image = image.encode('utf-8')
+                except AttributeError:
+                    pass
+
+                append = {'title': title, 'url': url, 'image': image}
+                if next != '':
+                    append['next'] = next
+                self.list.append(append)
+            except BaseException:
+                pass
+
+        try:
+            u = [list(range(0, len(self.list)))[i:i+50] for i in list(range(len(list(range(0, len(self.list))))))[::50]]
+            u = [','.join([self.list[x]['url'] for x in i]) for i in u]
+            u = [self.content_link % i + self.key_link for i in u]
+
+            threads = []
+            for i in list(range(0, len(u))):
+                threads.append(workers.Process(self.thread, u[i], i))
+                self.data.append('')
+            [i.start() for i in threads]
+            [i.join() for i in threads]
+
+            items = []
+            for i in self.data:
+                items += json.loads(i)['items']
+        except BaseException:
+            pass
+
+        for item in list(range(0, len(self.list))):
+            try:
+                vid = self.list[item]['url']
+
+                self.list[item]['url'] = self.play_link.format(vid)
+
+                d = [(i['id'], i['contentDetails']) for i in items]
+                d = [i for i in d if i[0] == vid]
+                d = d[0][1]['duration']
+
+                duration = 0
+                try:
+                    duration += 60 * 60 * int(re.findall('(\d*)H', d)[0])
+                except BaseException:
+                    pass
+                try:
+                    duration += 60 * int(re.findall('(\d*)M', d)[0])
+                except BaseException:
+                    pass
+                try:
+                    duration += int(re.findall('(\d*)S', d)[0])
+                except BaseException:
+                    pass
+                duration = str(duration)
+
+                self.list[item]['duration'] = duration
+            except BaseException:
+                pass
+
+        return self.list
+
+    def thread(self, url, i):
+
+        try:
+            result = requests.get(url).text
+            self.data[i] = result
+        except BaseException:
+            return
+
+    def play(self, name, url=None, as_script=True):
+
+        try:
+
+            url = self.worker(name, url)
+            if url is None:
+                return
+
+            title = control.infoLabel('listitem.title')
+            if title == '':
+                title = control.infoLabel('listitem.label')
+            icon = control.infoLabel('listitem.icon')
+
+            item = control.item(path=url, iconImage=icon, thumbnailImage=icon)
+
+            try:
+                item.setArt({'icon': icon})
+            except BaseException:
+                pass
+
+            item.setInfo(type='Video', infoLabels={'title': title})
+
+            if as_script:
```

Comment:
why do you need this? The addon is a plugin not a script. Plugins should only use `setResolvedUrl`

Reply:
ok.